logstash-core/src/main/java/org/logstash/execution/WorkerLoop.java

```diff
@@ -84,12 +84,15 @@ public void run() {
                 if (batch.filteredSize() > 0 || isFlush) {
                     consumedCounter.add(batch.filteredSize());
                     readClient.startMetrics(batch);
-                    final int outputCount = execution.compute(batch, isFlush, false);
-                    int filteredCount = batch.filteredSize();
-                    filteredCounter.add(filteredCount);
-                    readClient.addOutputMetrics(outputCount);
-                    readClient.addFilteredMetrics(filteredCount);
-                    readClient.closeBatch(batch);
+                    readClient.executeWithTimers(() -> {
```

Comment:
This change is a little absurd, but currently the metrics are owned by the `QueueReadClient` so it is not net-worse. I would like to pull the metrics up to the worker loop (or perhaps elsewhere) in a second pass, but wanted this change to be distinctly visible.